pkg/cmd/cli/secrets/link_secret_to_obj.go

```diff
@@ -0,0 +1,190 @@
+package secrets
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// LinkSecretRecommendedName `oc secrets link`
+	LinkSecretRecommendedName = "link"
+
+	// AddSecretRecommendedName Maintained for backwards-compatibility, but deprecated
+	AddSecretRecommendedName = "add"
+
+	// TODO: move to examples
+	linkSecretLong = `
+Link secrets to a service account
+
+Linking a secret enables a service account to automatically use that secret for some forms of authentication`
+
+	linkSecretExample = `  # Add an image pull secret to a service account to automatically use it for pulling pod images:
+  %[1]s serviceaccount-name pull-secret --for=pull
+
+  # Add an image pull secret to a service account to automatically use it for both pulling and pushing build images:
+  %[1]s builder builder-image-secret --for=pull,mount
+
+  " If the cluster's serviceAccountConfig is operating with limitSecretReferences: True, secrets must be added to the pod's service account whitelist in order to be available to the pod:
+  %[1]s pod-sa pod-secret`
+)
+
+type LinkSecretOptions struct {
+	SecretOptions
+
+	ForMount bool
+	ForPull  bool
+
+	typeFlags []string
+}
+
+// NewCmdLinkSecret creates a command object for linking a secret reference to a service account
+func NewCmdLinkSecret(name, fullName string, f *kcmdutil.Factory, out io.Writer) *cobra.Command {
+	o := &LinkSecretOptions{SecretOptions{Out: out}, false, false, nil}
+
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s serviceaccounts-name secret-name [another-secret-name]...", name),
+		Short:   "Link secrets to a ServiceAccount",
+		Long:    linkSecretLong,
+		Example: fmt.Sprintf(linkSecretExample, fullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := o.Complete(f, args); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.Validate(); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.LinkSecrets(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+
+		},
+	}
+
+	cmd.Flags().StringSliceVar(&o.typeFlags, "for", []string{"mount"}, "type of secret to link: mount or pull")
+
+	return cmd
+}
+
+// NewCmdAddSecret creates a command object for linking a secret reference to a service account
```

Comment:
Instead of having two `NewCmd*` and creating two commands to keep backwards compatibility, you should use `moved(...)` from `cli.go` like in [here](https://github.com/sgallagher/origin/blob/09101fa194f3b2109e2c632b9cc2e18ba9a5b308/pkg/cmd/cli/cli.go#L181). It takes care of adding the "DEPRECATED" message, hiding the deprecated command in help, and allows you to only have one command built, `NewCmdLinkSecret`.